source/Halibut.Tests/Support/TryListenWebSocket.cs

```diff
@@ -0,0 +1,52 @@
+﻿using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Serilog;
+
+namespace Halibut.Tests.Support
+{
+    public class TryListenWebSocket
+    {
+        public static async Task<ListeningWebSocket> WebSocketListeningPort(ILogger logger, HalibutRuntime client, CancellationToken cancellationToken)
+        {
+            logger = logger.ForContext<TryListenWebSocket>();
+            for (int i = 0; i < 9; i++)
+            {
+                try
+                {
+                    return await WebSocketListeningPort(client, cancellationToken);
+                }
+                catch (Exception e)
```

Comment:
Could we catch a more specific exception just incase it's another error getting thrown?

Reply:
Catching System.Net.HttpListenerExceptio